File: ts_code/SingleNumber.ts
class SingleNumber {
    public static singleNumber(nums: number[]): number {
        let res = 0; // ②

        for (const n of nums) { // ③
            res ^= n; // ④
        }

        return res; // ⑤
    }
}

// Ejecución del código:
const number: number[] = [2, 2, 1];
const result = SingleNumber.singleNumber(number);
console.log("Número único:", result); // Resultado esperado: 1

File: python_code/SingleNumber.py
class SingleNumber:
    @staticmethod
    def single_number(nums):
        res = 0  # ②
        for n in nums:  # ③
            res ^= n  # ④
        return res  # ⑤

# Ejecución del código
nums = [2, 2, 1]
result = SingleNumber.single_number(nums)
print("Número único:", result)  # Resultado esperado: 1
